src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -461,35 +560,56 @@ async function getFlows(
   // - Subsequent fills after an initial partial fill.
   // - Slow fills.
   // - Fills that are considered "invalid" by the spoke pool client.
-  const fills: UbaFlow[] = await filterAsync(spokePoolClient.getFills(), async (fill: FillWithBlock) => {
-    const validWithinBounds =
-      fill.repaymentChainId === spokePoolClient.chainId &&
-      fill.fillAmount.eq(fill.totalFilledAmount) &&
-      fill.updatableRelayData.isSlowRelay === false &&
-      fill.blockNumber > (fromBlock as number) &&
-      fill.blockNumber < (toBlock as number);
-
-    const hasMatchingDeposit = (await resolveCorrespondingDepositForFill(fill, spokePoolClients)) !== undefined;
-
-    return validWithinBounds || hasMatchingDeposit;
-  });
-
-  const refundRequests: UbaFlow[] = await filterAsync(
-    spokePoolClient.getRefundRequests(fromBlock, toBlock),
-    async (refundRequest) => {
-      const result = await refundRequestIsValid(chainIdIndices, spokePoolClients, hubPoolClient, refundRequest);
-      if (!result.valid && logger !== undefined) {
-        logger.info({
-          at: "UBAClient::getFlows",
-          message: `Excluding RefundRequest on chain ${chainId}`,
-          reason: result.reason,
-          refundRequest,
-        });
-      }
-
-      return result.valid;
-    }
-  );
+  const fills: UbaFlow[] = (
+    await Promise.all(
+      spokePoolClient.getFills().map(async (fill: FillWithBlock): Promise<UbaFlow | undefined> => {
+        const validWithinBounds =
+          fill.repaymentChainId === spokePoolClient.chainId &&
+          fill.fillAmount.eq(fill.totalFilledAmount) &&
+          fill.updatableRelayData.isSlowRelay === false &&
+          fill.blockNumber >= (fromBlock as number) &&
+          fill.blockNumber <= (toBlock as number);
+        if (!validWithinBounds) {
+          return undefined;
+        } else {
+          const matchingDeposit = await resolveCorrespondingDepositForFill(fill, spokePoolClients);
+          if (matchingDeposit === undefined) {
+            return undefined;
+          } else {
+            return {
+              ...fill,
+              quoteBlockNumber: matchingDeposit.quoteBlockNumber,
+            };
+          }
+        }
```

Comment:
The ` } else {` statements here are redundant, and the code squashes down nicely if they're removed.

```suggestion
        if (!validWithinBounds) {
          return undefined;
        }

        const matchingDeposit = await resolveCorrespondingDepositForFill(fill, spokePoolClients);
        if (matchingDeposit === undefined) {
          return undefined;
        }

        return { ...fill, quoteBlockNumber: matchingDeposit.quoteBlockNumber };
```